src/bin/ion/commands/beta/symtab/filter.rs

```diff
@@ -0,0 +1,101 @@
+use crate::commands::{IonCliCommand, WithIonCliArgument};
+use anyhow::{bail, Context, Result};
+use clap::{Arg, ArgAction, ArgMatches, Command};
+use ion_rs::binary::non_blocking::raw_binary_reader::RawBinaryReader;
+use ion_rs::{IonReader, IonResult, IonType, SystemReader, SystemStreamItem};
+use memmap::MmapOptions;
+use std::fs::File;
+use std::io::{stdout, BufWriter, Write};
+
+pub struct SymtabFilterCommand;
+
+impl IonCliCommand for SymtabFilterCommand {
+    fn name(&self) -> &'static str {
+        "filter"
+    }
+
+    fn about(&self) -> &'static str {
+        // XXX Currently only supports binary input
+        "Filters user data out of a binary Ion stream, leaving only the symbol table(s) behind."
+    }
+
+    fn configure_args(&self, command: Command) -> Command {
+        command.with_input()
+            .with_output()
+            .arg(Arg::new("lift")
+                .long("lift")
+                .short('l')
+                .required(false)
+                .action(ArgAction::SetTrue)
+                .help("Remove the `$ion_symbol_table` annotation from symtabs, turning them into visible user data")
+            )
+    }
+
+    fn run(&self, _command_path: &mut Vec<String>, args: &ArgMatches) -> Result<()> {
+        let mut output: Box<dyn Write> = if let Some(output_file) = args.get_one::<String>("output")
+        {
+            let file = File::create(output_file).with_context(|| {
+                format!(
+                    "could not open file output file '{}' for writing",
+                    output_file
+                )
+            })?;
+            Box::new(BufWriter::new(file))
+        } else {
+            Box::new(stdout().lock())
+        };
+
+        let lift_requested = args.get_flag("lift");
+
+        if let Some(input_file_names) = args.get_many::<String>("input") {
+            // Input files were specified, run the converter on each of them in turn
+            for input_file in input_file_names {
+                let file = File::open(input_file.as_str())
+                    .with_context(|| format!("Could not open file '{}'", &input_file))?;
```

Comment:
Once again: https://github.com/amazon-ion/ion-cli/issues/55

I'm not saying we need to solve it here, but I feel like there's got to be a better way for consistent input/output handling through all the CLI subcommands.